### PR TITLE
Set i18next language using session object

### PR DESCRIPTION
**Pull request by brandones**

### CODE REVIEW

File: src/openmrs-esm-api.ts [hunk 1]
```diff
@@ -32,13 +32,9 @@ export {
...
 } from "./shared-api-objects/current-patient";
-export {
-  useCurrentPatient
-} from "./shared-api-objects/use-current-patient.hook";
+export { useCurrentPatient } from "./shared-api-objects/use-current-patient.hook";
 
-export {
-  default as UserHasAccessReact
-} from "./shared-api-objects/user-has-access-react.component";
+export { default as UserHasAccessReact } from "./shared-api-objects/user-has-access-react.component";
```
Comment (brandones) [inline src/openmrs-esm-api.ts:37]: Prettier. No real change.

File: src/shared-api-objects/current-user.ts [hunk 2]
```diff
@@ -20,13 +21,29 @@ function getCurrentUser(
...
     ),
     filter(Boolean)
   ) as Observable<LoggedInUser | UnauthenticatedUser>;
 }
 
+function setUserLanguage(sessionResponse) {
+  if (sessionResponse?.data?.user?.userProperties?.defaultLocale) {
+    const locale = sessionResponse.data.user.userProperties.defaultLocale;
+    //@ts-ignore
+    const i18nInstance = i18n.default || i18n;
```
Comment (FlorianRappl) [inline src/shared-api-objects/current-user.ts]: Alternatively, what we could do here is set the `lang` attribute of `documentElement` (assuming this is our top prio - coming from the SSR, too).

Then just calling `i18nInstance.changeLanguage()` is sufficient. Is is using the detector logic again.

This way we always use the same detector logic / order.

Comment (FlorianRappl) [reply src/shared-api-objects/current-user.ts]: Additional remark:

In the outlined case we don't even need to trigger `i18nInstance.changeLanguage()`  here. When setting up the `i18nInstance` we could install a mutation observer on the `documentElement` (no subtree), which triggers when attributes change.

In this case we would always be consistent.

Comment (FlorianRappl) [reply src/shared-api-objects/current-user.ts]: Remark:
For the empty `lng` argument of `changeLanguage` see https://github.com/i18next/i18next/blob/master/src/i18next.js#L270.

Comment (brandones) [reply src/shared-api-objects/current-user.ts]: > For the empty `lng` argument of `changeLanguage` see https://github.com/i18next/i18next/blob/master/src/i18next.js#L270.

I was going to ask where you found that! Definitely not in the docs. Good find.

Comment (brandones) [reply src/shared-api-objects/current-user.ts]: Using this and a mutation observer on the `documentElement` makes sense to me. I'll make the change.

Comment (FlorianRappl) [reply src/shared-api-objects/current-user.ts]: Awesome! Thanks :+1:!

File: src/shared-api-objects/current-user.ts [hunk 3]
```diff
@@ -20,13 +21,31 @@ function getCurrentUser(
...
 }
 
+function setUserLanguage(sessionResponse) {
+  if (sessionResponse?.data?.user?.userProperties?.defaultLocale) {
+    const locale = sessionResponse.data.user.userProperties.defaultLocale;
+    const htmlLang = document.documentElement.getAttribute("lang");
+    if (locale != htmlLang) {
+      document.documentElement.setAttribute("lang", locale);
+      //@ts-ignore
+      (i18n.default || i18n).changeLanguage().catch(e => {
```
Comment (FlorianRappl) [inline src/shared-api-objects/current-user.ts]: Ok so no mutation observer?

Should we have the whole language logic in one module and here only the adjustment of the attribute?

Comment (brandones) [reply src/shared-api-objects/current-user.ts]: Aw crap, I forgot that detail. Thanks for the reminder.  I'll set up the mutation observer.

Comment (brandones) [reply src/shared-api-objects/current-user.ts]: https://github.com/openmrs/openmrs-esm-root-config/pull/38

File: src/shared-api-objects/current-user.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { BehaviorSubject, Observable } from "rxjs";
 import { openmrsObservableFetch, FetchResponse } from "../openmrs-fetch";
-import { mergeAll, filter, map } from "rxjs/operators";
+import { mergeAll, filter, map, tap } from "rxjs/operators";
+import i18n from "i18next";
```
Comment (FlorianRappl) [inline src/shared-api-objects/current-user.ts]: Unused import, or?

Comment (brandones) [reply src/shared-api-objects/current-user.ts]: Fixed

**Review by FlorianRappl — APPROVED** [review]
LGTM!

**Pull request merged**